Automattic/newspack-blocks
# feat: author profile block

### All Submissions:






### Changes proposed in this Pull Request:



Registers a new "Author Profile" block which allows editors to choose either vanilla WP users or Co-authors Plus guest authors and display them in a standardized format in post or page content.

Also updates some NPM dependencies because I was having trouble with the latest version of `newspack-components` without updating, and many packages are very outdated. Please let me know if you see any weirdness in other blocks that may be a result of the updates.

Closes #723.

### How to test the changes in this Pull Request:

1. In a post, confirm there's a new "Author Profile" block under the Newspack category in the block inserter.
2. Insert an author profile block. Observe the search UI and confirm that the list of suggestions shows both regular WP users and Co-authors guest authors*:

<img width="816" alt="Screen Shot 2021-05-28 at 4 34 45 PM" src="https://user-images.githubusercontent.com/2230142/120047762-a491fb80-bfd2-11eb-909e-3f2bac444178.png">

3. Save the post without selecting an author. View on the front-end; confirm that the block renders nothing if no author is selected.
4. Back in the editor, select an author from the suggestions list and by typing a name into the autocomplete input. (Use the "clear" button to choose another author after selecting one.) Add a second Author Profile block and make sure one contains a regular WP user and the other a guest author. Confirm that the preview shows the author info in both cases as expected**:

<img width="822" alt="Screen Shot 2021-05-28 at 4 57 43 PM" src="https://user-images.githubusercontent.com/2230142/120048986-e53f4400-bfd5-11eb-9117-e8b8c9855810.png">

5. Save the post, view on the front-end. Confirm that both blocks render as expected( also test at smaller viewports to confirm that the styles collapse with the avatar on top of the bio info):

<img width="805" alt="Screen Shot 2021-05-28 at 4 57 34 PM" src="https://user-images.githubusercontent.com/2230142/120048971-e1132680-bfd5-11eb-8ae2-1a00ea7fe260.png">

6. Back in the editor, play with the various attribute options. Confirm that both the editor preview and the front-end render as expected based on the options selected.

<img width="279" alt="Screen Shot 2021-05-28 at 4 40 10 PM" src="https://user-images.githubusercontent.com/2230142/120048060-621cee80-bfd3-11eb-80ca-f2640f37ca1c.png">

7. Disable the Co-authors Plus plugin. On the front-end, confirm that blocks containing guest authors now render nothing. In the editor, confirm that these blocks show a "not found" warning:

<img width="816" alt="Screen Shot 2021-05-28 at 4 29 49 PM" src="https://user-images.githubusercontent.com/2230142/120048154-97294100-bfd3-11eb-8228-25c94f3caaaa.png">

### Notes

\* Because guest authors are a custom post type, we can't query both regular users and guest authors in a single query. Instead, we must query them separately and combine the results. For this reason, the list of suggestions will always alternate between the two types of authors and may not display strictly in reverse chronological order.

** The only _display_ difference between regular WP users and guest authors is in the social links: WP users have more metafield options for various social media services, whereas guest authors basically have just a website and email field (also Yahoo, AIM, and Jabber, but these aren't used elsewhere in the theme so I didn't include handling for them). WP users and guest authors also differ in the format of their edit links and archive page links.

### Other information:
